checks/binary_artifact.go

```diff
@@ -24,18 +24,39 @@ import (
 // CheckBinaryArtifacts is the exported name for Binary-Artifacts check.
 const CheckBinaryArtifacts string = "Binary-Artifacts"
 
-//nolint
+// nolint
 func init() {
 	supportedRequestTypes := []checker.RequestType{
 		checker.CommitBased,
 	}
-	if err := registerCheck(CheckBinaryArtifacts, BinaryArtifacts, supportedRequestTypes); err != nil {
+	b := BinaryArtifactsCheck{}
+	if err := registerCheckInterface(CheckBinaryArtifacts, b, supportedRequestTypes); err != nil {
 		// this should never happen
 		panic(err)
 	}
 }
 
+type BinaryArtifactsCheck struct{}
+
+func (bac BinaryArtifactsCheck) Run(c *checker.CheckRequest) checker.CheckResult {
+	// your implementation here
+	rawData, err := raw.BinaryArtifacts(c.RepoClient)
+	if err != nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, err.Error())
+		return checker.CreateRuntimeErrorResult(CheckBinaryArtifacts, e)
+	}
+
+	// Return raw results.
+	if c.RawResults != nil {
+		c.RawResults.BinaryArtifactResults = rawData
+	}
+
+	// Return the score evaluation.
+	return evaluation.BinaryArtifacts(CheckBinaryArtifacts, c.Dlogger, &rawData)
+}
+
 // BinaryArtifacts  will check the repository contains binary artifacts.
+// This is being used by AllStar and that is the reason to keep it.
```

Comment:
Sorry for any confusion on my initial comment, but I don't think we need two implementations, just maintaining the function signature

This could just be simplified as 
```
func BinaryArtifacts(c *checker.CheckRequest) checker.CheckResult {
    b := checks.BinaryArtifactsCheck{}
    result := b.Run(&req)
    return result
}
```

Reply:
👍